data-prepper-core/src/test/java/org/opensearch/dataprepper/acknowledgements/DefaultAcknowledgementSetTests.java

```diff
@@ -0,0 +1,215 @@
+            defaultAcknowledgementSet.acquire(handle);
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(false));
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(false));
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(true));
+        } catch (Exception e) {}
+    }
+
+    @Test
+    void testDefaultAcknowledgementAcquireError() {
+        defaultAcknowledgementSet.add(event);
+        DefaultEventHandle handle = (DefaultEventHandle) event.getEventHandle();
+        assertThat(handle, not(equalTo(null)));
+        assertThat(handle.getAcknowledgementSet(), equalTo(defaultAcknowledgementSet));
+        try {
+            defaultAcknowledgementSet.acquire(handle);
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(true));
+        } catch (Exception e) {}
+        assertThrows(RuntimeException.class,  () -> defaultAcknowledgementSet.acquire(handle));
+    }
+
+    @Test
+    void testDefaultAcknowledgementReleaseError() {
+        defaultAcknowledgementSet.add(event);
+        DefaultEventHandle handle = (DefaultEventHandle) event.getEventHandle();
+        assertThat(handle, not(equalTo(null)));
+        assertThat(handle.getAcknowledgementSet(), equalTo(defaultAcknowledgementSet));
+        assertThrows(RuntimeException.class,  () -> defaultAcknowledgementSet.release(handle, true));
+    }
+
+    @Test
+    void testDefaultAcknowledgementDuplicateReleaseError() {
+        defaultAcknowledgementSet.add(event);
+        DefaultEventHandle handle = (DefaultEventHandle) event.getEventHandle();
+        assertThat(handle, not(equalTo(null)));
+        assertThat(handle.getAcknowledgementSet(), equalTo(defaultAcknowledgementSet));
+        try {
+            defaultAcknowledgementSet.acquire(handle);
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(true));
+        } catch (Exception e) {}
+        assertThrows(RuntimeException.class,  () -> defaultAcknowledgementSet.release(handle, true));
+    }
+
+    @Test
+    void testDefaultAcknowledgementSetWithCustomCallback() {
+        defaultAcknowledgementSet = createObjectUnderTestWithCallback(
+            (flag) -> {
+                acknowledgementSetResult = flag;
+            }        
+        );
+        defaultAcknowledgementSet.add(event);
+        DefaultEventHandle handle = (DefaultEventHandle) event.getEventHandle();
+        assertThat(handle, not(equalTo(null)));
+        assertThat(handle.getAcknowledgementSet(), equalTo(defaultAcknowledgementSet));
+        try {
+            defaultAcknowledgementSet.acquire(handle);
+            assertThat(defaultAcknowledgementSet.release(handle, true), equalTo(true));
+        } catch (Exception e) {}
+
+        while (!defaultAcknowledgementSet.isDone()) {
+            try {
+                Thread.sleep(1000);
+            } catch (Exception e){}
```

Comment:
Rather than suppress this exception, you can add `throws InterruptedException` to the test method. Then you can remove the `try-catch` entirely.